(Partial) removal of WrapperProps (part 2)

dimitropoulos opened the pull request:
This PR relates to (but does not close) the wrapper props ticket, INS-1058.  This continues the work started in https://github.com/Kong/insomnia/pull/4431.

Specifically, it focuses on the data (not callbacks) that have (or can/should have) simple state selectors.

This severs the tie at the root, where the props are first passed.  Yes, ultimately this will mean more work for us _overall_, but, as the previous PR and now this PR demonstrate, this is a gigantic job.  I don't want to change too much all at once.  The changes done this way in this PR are relatively safe, comparatively, since we're using (in most cases) exactly the same selector that Wrapper or App used in the first place.  Where there weren't selectors in those cases, I created some using exactly the same logic.

I don't agree, by the way, with some of the decisions and fallbacks in some of these selectors, but this is no time for scope creep.  It's already quite complicated.  Once we have it all wired up a little better later, we can address the fallbacks.

I would encourage a commit-by-commit review.  I tried to keep all the changes mostly atomic so that it'd be easier to reason about what happened in this PR.

dimitropoulos committed:
moves selector-able WrapperProps in WrapperHome

dimitropoulos committed:
moves selector-able WrapperProps in WrapperUnitTest

dimitropoulos committed:
creates selectors for remaining wrapperProps without selectors

dimitropoulos committed:
fixes "all" selector names

these were the only two in the project that were misnamed with respect to the state naming.  now, they match (like all others)

dimitropoulos committed:
moves selector-able WrapperProps in WrapperDebug

gatzjames commented on packages/insomnia-app/app/ui/components/wrapper-debug.tsx:
```diff
@@ -326,3 +328,24 @@ export class WrapperDebug extends PureComponent<Props> {
+  sidebarFilter: selectSidebarFilter(state),
+});
+
+export const WrapperDebug = connect(mapStateToProps)(UnconnectedWrapperDebug);
```
Not a fun of the Unconnected____ naming convention but it's similar in other places so something to consider in general about how we name these components.

dimitropoulos replied:
I actually hate it.  Here's why.  If you think about it.... the thing that's actually unconnected is (in this case) `WrapperDebug`.  The thing we call `UnconnectedWrapperDebug` IS CONNECTED.  THAT"S THE WHOLE POINT.

I digress.  You already know, though.  It's no time to get creative and rock the boat.  ALL of these "UnconnectedXyz" components are going to be gone as soon as possible, anyway, and replaced with hooks components.

gatzjames commented on packages/insomnia-app/app/ui/redux/sidebar-selectors.ts:
```diff
@@ -42,19 +43,42 @@ export interface SidebarChildren {
   selectEntitiesChildrenMap,
-  (collapsed, pinned, activeWorkspace, activeWorkspaceMeta, childrenMap): SidebarChildren => {
+  selectSidebarFilter,
+  (collapsed, pinned, activeWorkspace, childrenMap, sidebarFilter): SidebarChildren => {
```
😱 TIL that's quite a big selector. Keeping an eye on it to inform future database/cache etc. options

dimitropoulos replied:
my first time seeing it as well.  quite nasty.

gatzjames reviewed (approved):
Nice and clean! 🚀👏

dimitropoulos committed:
bugfix: worspaces was never renamed since projects were introduced

dimitropoulos committed:
workspaces always === workspaces, same for workspacesForActiveProject

there was a bug fix in the previous commit because of technical debt from when we rearchitected the app for projects.  this fixes that problem from ever happening again.  there's a big difference between "workspaces" and "workspacesForActiveProject", yet the app called them the same thing in different places.

dimitropoulos committed:
makes activeActivity wrapper prop match name in state

after the bug two commits ago, it seemed appropriate to fix exactly the same thing here.

dimitropoulos committed:
remove duplicated variables for all state selectors

dimitropoulos committed:
removes last name mismatch: "workspaceChildren"

dimitropoulos commented:
hey @gatzjames I found one bug and fixed it (as well as the root cause for 2 others that had not yet bugged but were like it).  I'm quite confident it's still ready to go, but please take another peek just in case.

gatzjames reviewed (approved):
🚀
